tests/test_search.py

```diff
@@ -1,60 +1,205 @@
+        for x in test_matches:
+            create = x - indexed
+            indexed = x
+            bundle_ids.extend(self.populate_search_index(self.index_document, create))
+            with self.subTest("Search Returns %i Matches When %i Documents Indexed.".format(x, x)):
+                self.verify_search_results(smartseq2_paired_ends_query, x, bundle_ids)
+
+    def test_elasticsearch_exception(self):
+        # Test Elasticsearch exception handling by setting an invalid endpoint
+        es_logger = logging.getLogger("elasticsearch")
+        original_es_level = es_logger.getEffectiveLevel()
+        original_es_endpoint = os.getenv("DSS_ES_ENDPOINT", "localhost")
+
+        es_logger.setLevel("ERROR")
+        try:
+            os.environ['DSS_ES_ENDPOINT'] = "bogus"
+            response = self.assertPostResponse(
+                "/v1/search",
+                json_request_body=smartseq2_paired_ends_query,
+                expected_code=requests.codes.internal_server_error)
+            self.assertEqual(response.json['code'], "elasticsearch_error")
+            self.assertEqual(response.json['title'], "Elasticsearch operation failed")
+        finally:
+            os.environ['DSS_ES_ENDPOINT'] = original_es_endpoint
+            es_logger.setLevel(original_es_level)
+
+    def populate_search_index(self, index_document: dict, count: int) -> list:
+        es_client = ElasticsearchClient.get(logger)
+        bundles = []
+        for i in range(count):
+            bundle_uuid = str(uuid.uuid4())
+            version = get_version()
+            index_document['manifest']['version'] = version
+            bundle_id = f"{bundle_uuid}.{version}"
+            bundle_url = f"http://localhost/v1/bundles/{bundle_uuid}?version={version}"
+
+            es_client.index(index=DSS_ELASTICSEARCH_INDEX_NAME,
+                            doc_type=DSS_ELASTICSEARCH_DOC_TYPE,
+                            id=bundle_id,
+                            body=index_document)
+            bundles.append((bundle_id, bundle_url))
+        return bundles
+
+    def verify_search_results(self, query, expected_result_length=0, bundles=[]):
+        timeout = 5
+        timeout_time = timeout + time.time()
+        while time.time() <= timeout_time:
+            response = self.assertPostResponse(
+                "/v1/search",
+                json_request_body=(query),
+                expected_code=requests.codes.ok)
+            search_response = response.json
+            if len(search_response['results']) == expected_result_length:
+                break
+            elif len(search_response['results']) > expected_result_length:
+                self.fail("elasticsearch more results than expected.")
+            else:
+                time.sleep(0.5)
+        else:
+            self.fail("elasticsearch failed to return all results.")
+
+        self.assertDictEqual(search_response['query'], query)
+        self.assertEqual(len(search_response), 2)
```

Comment:
is this a requirement?

Reply:
It's not a requirements written anywhere, but it is the current behavior of our code.